tests/base/src/test/java/org/keycloak/tests/admin/group/GroupTest.java

```diff
@@ -212,219 +201,180 @@ public void doNotAllowSameGroupNameAtSameLevel() throws Exception {
+    public void createAndTestGroups() {
+        RoleRepresentation topRole = createRealmRole(managedRealm, RoleBuilder.create().name("topRole").build());
+        RoleRepresentation level2Role = createRealmRole(managedRealm, RoleBuilder.create().name("level2Role").build());
+        RoleRepresentation level3Role = createRealmRole(managedRealm, RoleBuilder.create().name("level3Role").build());
 
         // Role events tested elsewhere
-        assertAdminEvents.clear();
+        adminEvents.skipAll();
+        RealmResource realm = managedRealm.admin();
 
         GroupRepresentation topGroup = new GroupRepresentation();
         topGroup.setName("top");
-        topGroup = createGroup(realm, topGroup);
+        String topGroupId = ApiUtil.getCreatedId(realm.groups().add(topGroup));
+        topGroup.setId(topGroupId);
+        AdminEventAssertion.assertEvent(adminEvents.poll(), OperationType.CREATE, AdminEventPaths.groupPath(topGroupId), topGroup, ResourceType.GROUP);
 
-        List<RoleRepresentation> roles = new LinkedList<>();
-        roles.add(topRole);
+        List<RoleRepresentation> roles = new LinkedList<>(List.of(topRole));
         realm.groups().group(topGroup.getId()).roles().realmLevel().add(roles);
-        assertAdminEvents.assertEvent(testRealmId, OperationType.CREATE, AdminEventPaths.groupRolesRealmRolesPath(topGroup.getId()), roles, ResourceType.REALM_ROLE_MAPPING);
+        AdminEventAssertion.assertEvent(adminEvents.poll(), OperationType.CREATE, AdminEventPaths.groupRolesRealmRolesPath(topGroup.getId()), roles, ResourceType.REALM_ROLE_MAPPING);
 
         GroupRepresentation level2Group = new GroupRepresentation();
         level2Group.setName("level2");
         Response response = realm.groups().group(topGroup.getId()).subGroup(level2Group);
-        response.close();
-        assertAdminEvents.assertEvent(testRealmId, OperationType.CREATE, AdminEventPaths.groupSubgroupsPath(topGroup.getId()), level2Group, ResourceType.GROUP);
-
-        URI location = response.getLocation();
+        AdminEventAssertion.assertEvent(adminEvents.poll(), OperationType.CREATE, AdminEventPaths.groupSubgroupsPath(topGroup.getId()), level2Group, ResourceType.GROUP);
         final String level2Id = ApiUtil.getCreatedId(response);
+
         final GroupRepresentation level2GroupById = realm.groups().group(level2Id).toRepresentation();
         assertEquals(level2Id, level2GroupById.getId());
         assertEquals(level2Group.getName(), level2GroupById.getName());
 
-        URLAssert.assertGetURL(location, adminClient.tokenManager().getAccessTokenString(), new URLAssert.AssertJSONResponseHandler() {
-            @Override
-            protected void assertResponseBody(String body) throws IOException {
-                GroupRepresentation level2 = JsonSerialization.readValue(body, GroupRepresentation.class);
-                assertEquals(level2Id, level2.getId());
-            }
-        });
-
```

Comment:
I got rid of these assertions on an access token. Is that fine? @stianst

Reply:
I misunderstood what this code does. Added it again here:
https://github.com/keycloak/keycloak/blob/502d2e82f23736f92a92cdb9e1bf6f8386fdec9c/tests/base/src/test/java/org/keycloak/tests/admin/group/GroupTest.java#L355-L361